### PR TITLE
[feat]: [CDS-76596]: Added Some unit test

### PR DESCRIPTION
**Pull request by Rakshitharness**
## Describe your changes
Adding Unit test for Class ApprovalInstanceServiceImpl, Get Method
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52643)

### CODE REVIEW

**Review by abhinavmittal-harness — COMMENTED** [review]
> Reviewed 2 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52643)* status: 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java` line 12 at r1](https://reviewable.io/reviews/harness/harness-core/52643#-Ndp2G7Y64RfRPBe5EUU:-Ndp2G7Y64RfRPBe5EUV:b-o5tz7g) ([raw file](https://github.com/harness/harness-core/blob/67f36b50cdc80b5ba7e15aee039a27f9d10b802b/pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java#L12)):*
> ```Java
> import static io.harness.annotations.dev.HarnessTeam.CDC;
> import static io.harness.annotations.dev.HarnessTeam.PIPELINE;
> import static io.harness.rule.OwnerRule.*;
> ```

You might want to update you intellij settings not to use start imports otherwise some of the pr checks will fail
___
*[`pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java` line 961 at r1](https://reviewable.io/reviews/harness/harness-core/52643#-Ndp2irfCuubPqJ_6nqP:-Ndp2irfCuubPqJ_6nqQ:bwj7t75) ([raw file](https://github.com/harness/harness-core/blob/67f36b50cdc80b5ba7e15aee039a27f9d10b802b/pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java#L961)):*
> ```Java
>     assertThatThrownBy(
>         () -> { approvalInstanceServiceImpl.findLatestApprovalInstanceByPlanExecutionIdAndType("", null); })
>         .isInstanceOf(InvalidArgumentsException.class)
> ```

create a new ut method for this

**Comment by yogesh-chauhan** [comment]
*[`pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java` line 172 at r1](https://reviewable.io/reviews/harness/harness-core/52643#-NdpUfVC-3Z6vmtDZ_sG:-NdpUfVC-3Z6vmtDZ_sH:boru16l) ([raw file](https://github.com/harness/harness-core/blob/67f36b50cdc80b5ba7e15aee039a27f9d10b802b/pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java#L172)):*
> ```Java
>   @Category(UnitTests.class)
>   public void testgetHarnessApprovalInstanceException() {
>     when(approvalInstanceRepository.findById(any())).thenReturn(Optional.empty());
> ```

if we know what is the method arguments, lets avoid using any() matchers
For eg. here, you know that the id is "Invalid"

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 2 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52643)* status: 1 file(s) to review. 0 of 2 approvals obtained.

**Comment by yogesh-chauhan** [comment]
@Rakshitharness , lets update PR description. Eg. can be what tests are you adding for which class etc ?

**Review by Abhishek-CDC — APPROVED** [review]

**Comment by Rakshitharness** [comment]
trigger uts

File: pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java [hunk 1]
```diff
@@ -146,6 +147,16 @@ public void testget() {
 
     assertThat(approvalInstanceServiceImpl.get("hello")).isEqualTo(entity);
   }
+  @Test
+  @Owner(developers = RAKSHIT_AGARWAL)
+  @Category(UnitTests.class)
+  public void testgetException() {
+    String Instance = "Invalid";
```
Comment (puthrayaharness) [inline pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java:154]: we do not initialize variables by capital letter. Please use `instance` instead.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java [hunk 2]
```diff
@@ -146,6 +147,16 @@ public void testget() {
 
     assertThat(approvalInstanceServiceImpl.get("hello")).isEqualTo(entity);
   }
+  @Test
+  @Owner(developers = RAKSHIT_AGARWAL)
+  @Category(UnitTests.class)
```
Comment (naman-goenka) [inline pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java:152]: method name should be `testGetException`

File: pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java [hunk 3]
```diff
@@ -160,6 +171,17 @@ public void testgetHarnessApprovalInstance() {
     assertThat(approvalInstanceServiceImpl.getHarnessApprovalInstance("hello")).isEqualTo(entity);
   }
 
+  @Test
+  @Owner(developers = RAKSHIT_AGARWAL)
+  @Category(UnitTests.class)
+  public void testgetHarnessApprovalInstanceException() {
```
Comment (naman-goenka) [inline pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java:177]: Similarly here

File: pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java [hunk 4]
```diff
@@ -146,6 +147,16 @@ public void testget() {
 
     assertThat(approvalInstanceServiceImpl.get("hello")).isEqualTo(entity);
   }
+  @Test
+  @Owner(developers = RAKSHIT_AGARWAL)
+  @Category(UnitTests.class)
+  public void testgetException() {
+    String instance = "Invalid";
+    when(approvalInstanceRepository.findById(instance)).thenReturn(Optional.empty());
+    assertThatThrownBy(() -> { approvalInstanceServiceImpl.get(""); })
```
Comment (naman-goenka) [inline pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/step/ApprovalInstanceServiceTest.java:156]: nit : you can also reuse `APPROVAL_ID` for `instance`

**Review by naman-goenka — COMMENTED** [review]
Added comments

**Comment by Rakshitharness** [comment]
trigger uts

**Comment by Rakshitharness** [comment]
trigger messagecheck

**Comment by Rakshitharness** [comment]
trigger uts

**Review by puthrayaharness — APPROVED** [review]

**Review by Abhishek-CDC — APPROVED** [review]

**Comment by Abhishek-CDC** [comment]
trigger review

**Pull request merged**